### ISSUE TITLE
App Crash on iOS 15 Beta 5

### ISSUE THREAD

**Issue by sso-bitwarden** [bitwarden/mobile#1493]
I am able to log in. Regardless of whether or not I make any changes to the settings (enable FaceID or not, change timeouts, etc.) the app will immediately close out as soon as I reopen the app. An ips is generated and the stack  racestarts with (if it helps):

 "exception" : {"codes":"0x0000000000000000, 0x0000000000000000","rawCodes":[0,0],"type":"EXC_CRASH","signal":"SIGABRT"},
  "asi" : {"libsystem_c.dylib":["abort() called"]},
  "lastExceptionBacktrace" : [{"imageOffset":628456,"imageIndex":5},{"imageOffset":89956,"imageIndex":6},{"imageOffset":1532244,"imageIndex":5},{"imageOffset":188548,"imageIndex":5},{"imageOffset":184768,"imageIndex":5},{"imageOffset":1792184,"imageIndex":11},{"imageOffset":767644,"imageIndex":5},{"imageOffset":836336,"imageIndex":5},{"imageOffset":23652,"imageIndex":5},{"imageOffset":46084,"imageIndex":5},{"imageOffset":126920,"imageIndex":5},{"imageOffset":5004,"imageIndex":8},{"imageOffset":5340688,"imageIndex":9},{"imageOffset":2713072,"imageIndex":9},{"imageOffset":5973944,"imageIndex":3},{"imageOffset":5262784,"imageIndex":3},{"imageOffset":5262660,"imageIndex":3},{"imageOffset":595272,"imageIndex":3},{"imageOffset":4415840,"imageIndex":3},{"imageOffset":284552,"imageIndex":4},{"imageOffset":1055304,"imageIndex":4},{"imageOffset":1080480,"imageIndex":4},{"imageOffset":143556,"imageIndex":4},{"imageOffset":34310384,"imageIndex":3},{"imageOffset":595088,"imageIndex":3},{"imageOffset":105252,"imageIndex":10}],
  "faultingThread" : 0,
  "threads" : [{"triggered":true,"id":371738,"threadState":{"x":[{"value":0},{"value":0},{"value":0},{"value":0},{"value":0},{"value":10000000},{"value":5120},{"value":18446744069414586880},{"value":9007292605623196026},{"value":9007292609895505146},{"value":1000},{"value":12532993},{"value":16393},{"value":4763844608},{"value":16383},{"value":33922},{"value":328},{"value":4345218432},{"value":0},{"value":6},{"value":259},{"value":4345218656},{"value":4749022304},{"value":4764860752},{"value":4763726336},{"value":4379446488},{"value":4764860928},{"value":5},{"value":4309885279}],"flavor":"ARM_THREAD_STATE64","lr":{"value":8390571060},"cpsr":{"value":1073741824},"fp":{"value":6161400176},"sp":{"value":6161400144},"esr":{"value":1442840704,"description":" Address size fault"},"pc":{"value":7425071544,"matchesCrashFrame":1},"far":{"value":6165807040}},"name":"tid_103","queue":"com.apple.main-thread","frames":[{"imageOffset":31160,"imageIndex":0},{"imageOffset":29748,"imageIndex":1},

## Environment

- Operating system:  iOS 15 Beta 5
- Build Version: 2.11.1 (edited)

**Comment by jerryc127** [comment]
yes. I update to iOS 15 beta 5 
The app cannot open when click the icon directly 
but it can use to fill in the password with keyboard

**Comment by the-great-emu-kerfuffle** [comment]
I am the reporter of this bug. More info of questionable quality: If I leave the app running, and do not force close, I can continue to access the app with no issues. If I close it, I have to uninstall & reinstall and then reauthenticate to use it. Also, I can offload the app and reinstall and it will work, which is quicker. 

I will need to see if the problem exists on the latest  iPadOS beta.

I’m willing to assist in troubleshooting or providing more information but I am not an iOS developer by practice.

**Comment by the-great-emu-kerfuffle** [comment]
App version is 2.11.1, the latest in the App Store. Not 2.11.3.

**Comment by nicopasla** [comment]
Same problem for me since today, restart help but temporarily

**Comment by trparky** [comment]
I have the same issue.

**Comment by jkelol111** [comment]
I can confirm the same issue on iOS 15 Beta 5, crashes when tapping icon on Home Screen/App Library as well as keyboard autofill.

**Comment by evlsmurf** [comment]
Can confirm this is happening on iPad Pro iOS15 B5 as well but a simple shutdown from settings and turning back on will allow you to use bitwarden again until you close out from multitasking.

**Comment by ax-meyer** [comment]
Same behavior here. 
The first major bug I experience on the beta, and directly something with a big impact. Love my luck.

**Comment by josephazrak** [comment]
Can confirm I am facing the same issue on 15.0 (19A5318f) (beta 5). App crashes on open--rather high impact.

**Comment by sleeping10** [comment]
Workaround for the moment if autofill stops working in safari use firefox works fine even when the app crash and if you need to access the app reinstall the app

**Comment by trparky** [comment]
I have submitted some feedback in the Apple Feedback app, apparently more than ten of you have submitted similar feedback to Apple. Keep it going guys, maybe it's something in the OS that's causing this issue.

**Comment by Jeehut** [comment]
Same problem here. As a workaround I'm now using vault.bitwarden.com which is not designed for Mobile, but does the job temporarily.

Just in case someone forgot. 😉

**Comment by m4rkv5** [comment]
@Jeehut My app runs for multiple hours when i restart my phone.

**Comment by josephazrak** [comment]
> @Jeehut On my phone bitwarden runs for multiple hours after a reboot.

That makes sense--it's the time it takes for Jetsam to kill the app in the background naturally. The fact remains that once the app is closed, it will crash on reopen until the next reboot.

**Comment by trparky** [comment]
Any updates on this issue from the devs? Are we close to seeing a fix for this perhaps deployed via Test Flight? Or is this an iOS 15-related bug?

**Comment by kkankala** [comment]
Any update regarding the fix?

**Comment by atridadl** [comment]
Seems like an odd one to keep quiet on considering the impact it has.

**Comment by TheJulianJES** [comment]
It's possible that https://github.com/bitwarden/mobile/pull/1495 fixes this

**Comment by takov751** [comment]
So far it's similar to the twitter app, however that app on the second time runs . I would be really interested what changed brought this behaviour

**Comment by tdiehl** [comment]
Seems to be affecting lots of Xamarin-based apps (including one I work on). See also xamarin/xamarin-macios#12427.

**Comment by Wxfisch** [comment]
I’ve seen the same behavior. Though I have not been able to use app fill at all until I reboot at which point as others have said it works for some time. 
I was directed to this one dude by support and find it odd that it’s not been assigned or even acknowledged by the devs really.

**Comment by TheModdersDen** [comment]
Same issue here. iOS build 15.0 (19A5318f), iPhone 12 Pro.

I can temporarily resolve the issue by offloading the app and then re-downloading it, but can only keep it open for a short amount of time, as noted by others before it begins to crash again.

Uninstalling it and reinstalling it has the same effect, but again, not ideal.

Just thought I’d pitch my 2 cents here…

**Comment by nicknitewolf** [comment]
![27885118-9DB2-4D5A-9465-5FA84AA8B443](https://user-images.githubusercontent.com/6513413/129651048-15830b9d-729c-439d-bd20-dd9d65d00f0f.jpeg)

iOS 15 Beta bug. It will be fixed in the next beta…

**Comment by mattpowell19** [comment]
Has there been any movement on this?
I know that it is a free app, but at the same time would be ideal if this would work. Love the app but getting frustrated that will not populate through iOS pop but instead have to go through web browser and copy password.

**Comment by trparky** [comment]
So I guess that this may actually be a bug in iOS 15 Beta 5. Now let's hope that beta 6 is released this week and it contains the fix.

**Comment by AlexHunterCodes** [comment]
Polite reminder to everyone: stating you're also running the same version on the same beta and are also affected by this issue in the same way, or saying you're affected without posting any diagnostically useful or new information (like version numbers, and which beta branch and number you're running) doesn't help developers fix the issue any faster. It's just creating more noise they have to sift through. If your comment boils down to just saying "+1" use GitHub's emoji reactions to leave a 👍 on an existing comment, instead of cluttering the thread.

The issue already has a reliable repro on Beta 5, and appears to be an iOS issue that's affecting other Xamarin apps too, that Apple is already aware of and will likely be fixed in the next beta release, which developers will probably get early next week, and the public beta testers a couple of days later. 

So this is probably a case of just being patient, and remembering that if you don't want to deal with bugs and crashes, don't run beta software on your primary devices.

**Comment by TheJulianJES** [comment]
Can reproduce on iOS 15 Developer Beta 6 (released today).

**Comment by jerryc127** [comment]
Beta 6 still have this problem

**Comment by KaliFyre** [comment]
[BitwardeniOS-2021-08-18-105349.zip](https://github.com/bitwarden/mobile/files/7003777/BitwardeniOS-2021-08-18-105349.zip)
I have a stack trace if needed, Dev Beta 5, latest app store version. it stopped autofilling a while ago but still allowed app access... then it just stopped allowing app access. Device uptime is in the low double digits of days though. I'll update/reboot device and see if any more useful information can be gleaned from Dev Beta 6.

**Comment by whatstevedid** [comment]
Interesting post on the xamarin thread here:
https://github.com/xamarin/xamarin-macios/issues/12427#issuecomment-900899606

Changing the minimum OS version to be 14.0 in the info.plist stopped the crash for this user. 

Any way we could get a TestFlight version with this change to test?

**Comment by trparky** [comment]
Someone reported this...

Hi all.
I was facing a similar issue, but got it working.
I changed the minimum OS version to be 14.0 in the info.plist, and it doesn't crash on the 2nd load.

**Comment by AlexHunterCodes** [comment]
> Changing the minimum OS version to be 14.0 in the info.plist stopped the crash for this user.

App currently has a minimum version of 10.0, so 14.0 is a big jump. I don't know what version spread Bitwarden's iOS users have, but it's probably not palatable for the team to drop so many older devices all at once. Especially not for the sake of something that might only be a temporary workaround for an issue that will likely be fixed in a later iOS beta or the final release.

The earlier comment about Apple changing their dynamic linker to improve app launch times suggests that too. My guess raising the minimum version eliminates some buggy or incomplete legacy compatibility codepath, resolving the crash. 

> Any way we could get a TestFlight version with this change to test?

We should figure out if this fix actually 100% works for the Bitwarden app first. I don't have a recent enough Mac to run the latest Xcode beta, so can't build it myself to find out. Any takers?

**Comment by whatstevedid** [comment]
> We should figure out if this fix actually 100% works for the Bitwarden app first. I don't have a recent enough Mac to run the latest Xcode beta, so can't build it myself to find out. Any takers?

I have an M1 Macbook Air that I could use, though I don't have an Apple Developer account any more so not sure how much use I'd be?

**Comment by nicknitewolf** [comment]
> I have an M1 Macbook Air that I could use, though I don't have an Apple Developer account any more so not sure how much use I'd be?

Not sure if I am just unfamiliar with visual studio / xcode but I can't seem to get bitwarden building with a free developer account. I am starting to think it's impossible since some of the app entitlements bitwarden uses requires a paid dev account.

**Comment by whatstevedid** [comment]
Just an update on this - looks like my feedback report that I submitted to Apple has now has the status changed to ‘potential fix identified’ - suggesting this should be sorted in an upcoming release. 
![30152E05-BE6C-43EF-B3DF-0D90E28E5EBF](https://user-images.githubusercontent.com/89134154/129977969-b01afc17-4210-43ee-8aa3-f57ae8ff879a.png)

**Comment by dfuentes03** [comment]
> changed

Thanks for the share! This has been the most annoying bug during iOS 15 I've experienced other than the video scrubbing bug.

**Comment by eliykat** [comment]
Hi everyone, I don't have much to add to this thread other than to say @AlexHunterCodes' comment [here](https://github.com/bitwarden/mobile/issues/1493#issuecomment-900156970) sums up my understanding of the issue. We've had some preliminary reports that iOS 15 Beta 6 solves the issue, but this does not seem to be the case for everyone. (Or that it solves the issue, but only if the app extension is not enabled.)

We expect that this crash should be resolved by continued beta releases. We'll continue to monitor this. I agree that we probably wouldn't want to bump the minimum version from 10 -> 14 solely to fix a bug in a beta iOS release.

Thanks everyone for your patience and for the helpful information in this thread.

**Comment by AlexHunterCodes** [comment]
Just upgraded to public beta 6 and confirm that I am still experiencing the same behaviour as beta 5. IPS crash files show the same `Address size fault` for the above repro steps. Also tried various combinations of disabling the app extension, killing both Bitwarden and Safari, rebooting the device and still crashes on the 2nd launch. Wasn't personally able to reproduce the preliminary reports of a fix.

Although, I think I do glimpse a single frame of the master password prompt screen now that I didn't see on beta 5. Might just be changes in the launch animation or me seeing what I want to see.

**Comment by trparky** [comment]
Apparently one developer managed to be able to fix it by setting the minimum iOS version to version 11 and it fixed the issue.  Do you, @eliykat, would object to that?

**Comment by hampton8** [comment]
If this is indeed the case but the BW team didn't want to change the minimum version at all (which is totally fair enough considering this is a bug with beta software), a Bitwarden Test Flight release with just this alteration made would be extremely helpful for those of us currently running the iOS 15 beta and who are unable to use the app (beta 6 didn't fix it for me either).

**Comment by trparky** [comment]
Yeah, but who really is still on anything older than iOS 12? There's got to be some kind of data out there to indicate how many people are on older iOS versions.

**Comment by mraif13** [comment]
> Yeah, but who really is still on anything older than iOS 12? There's got to be some kind of data out there to indicate how many people are on older iOS versions.

According to this [report](https://www.nodesagency.com/os-version-statistics/) iOS: 12.x: 1.91%, anything below is 0%.

**Comment by saoud** [comment]
I agree a test flight release could solve the issue for us that are running the beta iOS and would be greatly appreciated.

**Comment by dfuentes03** [comment]
> I agree a test flight release could solve the issue for us that are running the beta iOS and would be greatly appreciated. 

Agreed... this has so far been the most intrusive bug since I've been on iOS 15 Dev Beta since day 1. I do pay for BitWarden to support the continued development of this application, however, this bug has me contemplating migrating all my stored items to another password manager for the time being... which ultimately can result in me switching over permanently.

**Comment by saoud** [comment]
> > I agree a test flight release could solve the issue for us that are running the beta iOS and would be greatly appreciated.
> 
> Agreed... this has so far been the most intrusive bug since I've been on iOS 15 Dev Beta since day 1. I do pay for BitWarden to support the continued development of this application, however, this bug has me contemplating migrating all my stored items to another password manager for the time being... which ultimately can result in me switching over permanently.

I was thinking about doing the same thing but it seems like a huge hassle when a TestFlight version with an updated min iOS version would be a very easy fix for the devs. I do also pay for the application and have been loving it so far.

LastPass was awful, I am not going to switch to dashlane so the thing I would be thinking about doing is using the iCloud password manager I just don't really want to migrate all my data over to that nor am I sure if there is an easy way to do so.

**Comment by karassm** [comment]
I confirm the bug is still present in Beta 6 (public beta, on iPhone 12). A TestFlight version is a very good idea.

**Comment by hampton8** [comment]
The app isn't working for me on beta 6. Crashes after a relaunch just as before.

**Comment by dfuentes03** [comment]
> > > I agree a test flight release could solve the issue for us that are running the beta iOS and would be greatly appreciated.
> > 
> > 
> > Agreed... this has so far been the most intrusive bug since I've been on iOS 15 Dev Beta since day 1. I do pay for BitWarden to support the continued development of this application, however, this bug has me contemplating migrating all my stored items to another password manager for the time being... which ultimately can result in me switching over permanently.
> 
> I was thinking about doing the same thing but it seems like a huge hassle when a TestFlight version with an updated min iOS version would be a very easy fix for the devs. I do also pay for the application and have been loving it so far.
> 
> LastPass was awful, I am not going to switch to dashlane so the thing I would be thinking about doing is using the iCloud password manager I just don't really want to migrate all my data over to that nor am I sure if there is an easy way to do so.

I'm right there with you on LastPass and Dashlane... 1Password was what I had in mind, especially since I get access to a free family account via my organization's business account. 

@eliykat What's blocking updating the minimum iOS version? I'm sure you guys have access to analytics and can easily pull up what iOS versions your users are on... Just to give you an example, even the iPhone 5S and iPad mini 2 support iOS 11 - both devices were released way back in 2013! I'm positive that you have very little, if any, users on iOS 10..

**Comment by dfuentes03** [comment]
> I confirm the bug is still present in Beta 6 (public beta, on iPhone 12). A TestFlight version is a very good idea.
> 
> Edit : the app seems to work but not the extension (tested on safari, chrome, etc)

Give it some time - it'll stop working as well 😭

**Comment by hampton8** [comment]
> Apparently one developer managed to be able to fix it by setting the minimum iOS version to version 11 and it fixed the issue.

Has anyone here actually been able to confirm this first hand, out of interest?

**Comment by pyrox82** [comment]
Hi,

for me, the app is working in iOS 15 Beta 6 if you disable the automatic fill (Settings -> Passwords). Then I can at least use the app and copy the credentials and past them manually. Hope to get it fixed soon.

**Comment by josephazrak** [comment]
> Hi,
> 
> 
> 
> for me, the app is working in iOS 15 Beta 6 if you disable the automatic fill (Settings -> Passwords). Then I can at least use the app and copy the credentials and past them manually. Hope to get it fixed soon.

This does not work for me. Disabling Autofill as you suggest still causes app to crash.

**Comment by pyrox82** [comment]
> > Hi,
> > for me, the app is working in iOS 15 Beta 6 if you disable the automatic fill (Settings -> Passwords). Then I can at least use the app and copy the credentials and past them manually. Hope to get it fixed soon.
> 
> This does not work for me. Disabling Autofill as you suggest still causes app to crash.

You are right. Sorry. As soon as I close the app it will not start again. :(

**Comment by kspearrin** [comment]
As suggested by some as a workaround, we've submitted an update to Apple on the TestFlight public track that bumps the minimum iOS version to v12. Once it is approved by Apple, we will post a link to access it from the TestFlight app.

**Comment by hampton8** [comment]
That's awesome Kyle, thank you so much!

**Comment by AlexHunterCodes** [comment]
To the brigading redditors who have been linked here, I'll restate my reminder from earlier:
- We already have reliable steps to reproduce the bug.
- We already know both betas 5 and 6 are affected.
- We already know other Xamarin.iOS apps are affected.
- We already know Apple is aware of the issue, and has identified a potential fix. If you're affected, please report it to Apple with the Feedback Hub, as more reports will draw more of Apple's attention towards fixing this. Nobody here can do anything with FB codes, so don't bother linking them here. If someone *does* have a status update from Apple on their FB ticket, or any relevant RADAR links, please share them.
- We already pretty sure Apple's switched to weekly beta releases, so beta 7 will be out early-mid next week.
- We have unconfirmed reports that switching to minimum SDK version 11.0 or 14.0 is a viable workaround if you're willing to build and sideload the app yourself. You can do this if you are registered in the paid Apple Developer Program, are running macOS Big Sur, and have Visual Studio and the latest Xcode beta installed.

If you don't have new information or haven't found a workaround, please use the emoji reactions on existing comments instead of cluttering developers' inboxes by repeating what we already know. If you find potentially helpful information from other affected projects elsewhere, please link it instead of just giving us a vague summary and expecting people to go hunt for it.

> I agree a test flight release could solve the issue for us that are running the beta iOS and would be greatly appreciated.

> Apparently one developer managed to be able to fix it by setting the minimum iOS version to version 11 and it fixed the issue. Do you, @eliykat, would object to that? It's only one number increment from the current minimum of iOS v10.

Apple have already shifted this beta to weekly releases and it's already Friday. By the time this change is agreed by the team, built, QA'd, pushed to TestFlight, reviewed and approved by Apple, and finally in beta testers hands early/mid next week... beta 7 will already be out and might've fixed the issue anyway.  Also, TestFlight is invite-only and has a user cap (and I think is closed/full atm?) so pushing an update would inherently only be able to help some of those affected who are in the know. Many (including myself) would be left out. 

> I do pay for BitWarden to support the continued development of this application, however, this bug has me contemplating migrating all my stored items to another password manager for the time being... which ultimately can result in me switching over permanently.

I pay for Bitwarden too, but I agree with @eliykat that this is a "wait and see" situation. Dogpiling on and pressuring them by threatening to leave for other password managers is unfair and unconstructive. This is an **opt-in beta where you should expect bugs that disrupt your workflow**, which is why everyone including Apple says you shouldn't run betas on your primary devices. You knew what you were signing up for, no one forced you to upgrade, and no one should expect Bitwarden to bend over backwards to support their app on an unsupported and constantly shifting version of the operating system when they don't want to. It's entitled and totally unreasonable.

Migrating to another password manager takes a huge investment of time and effort (and in this case cutting your nose off to spite your face). Do you know what would take less time and effort? Recognising that the current state of the iOS beta isn't suitable for your needs or workflow, and downgrading to iOS 14 stable (or even just to 15.0 beta 4) and restoring a backup. No really, it's pretty quick and painless nowadays.

**Comment by AlexHunterCodes** [comment]
> As suggested by some as a workaround, we've submitted an update to Apple on the TestFlight public track that bumps the minimum iOS version to v12. Once it is approved by Apple, we will post a link to access it from the TestFlight app.

Thanks, that's really appreciated. Will the TestFlight link just be for existing members to download the new release, or will the program open to new applicants too (can't find a public link)? Here's hoping Apple approve it before they release beta 7 with a fix on Monday or Tuesday morning. 😆

> I do pay for BitWarden to support the continued development of this application, however, this bug has me contemplating migrating all my stored items to another password manager for the time being... which ultimately can result in me switching over permanently.

I'm a paying customer too, but I think it's extremely sad that some users felt the need to strong-arm Bitwarden with "an offer they couldn't refuse". It's is unfair and unconstructive, and its just not how things are done in open source. The iOS 15 beta is opt-in and you should expect bugs that disrupt your workflow, which is why nobody recommends running betas on critical devices. You knew what you were signing up for, no one forced you to upgrade, and its unfair to demand developers support their app on a constantly shifting unsupported OS version when they don't have to. In this case it's nice that they did, but only after you threatened them, which is just kinda yucky.

Migrating to another password manager and cutting your nose off to spite your face takes a huge investment of time and effort. Do you know what would take less time and effort? Recognising that the current state of the iOS beta isn't suitable for your needs or workflow, and downgrading to iOS 14 stable (or even just to 15.0 beta 4) and restoring a backup. Apple's made downgrades and backup restores pretty painless nowadays.

**Comment by trparky** [comment]
> As suggested by some as a workaround, we've submitted an update to Apple on the TestFlight public track that bumps the minimum iOS version to v12. Once it is approved by Apple, we will post a link to access it from the TestFlight app.

Like @AlexHunterCodes said, I appreciate it. Getting a link posted to the Test Flight version of the app will be greatly appreciated.

**Comment by spencer-gre** [comment]
Hi!

Initially I was just unable to auto fill on the latest IOS Beta so I would resort to copying from the app. Now, the app crashes on launch. Glad to see that others are encountering same bug.

**Comment by trparky** [comment]
Someone mentioned this...

"release builds with LLVM work without having to modify the deployment target."

**Comment by hlafaille** [comment]
Same issue here, hopefully TestFlight gets approved soon. Just switched to BitWarden and I really like it, but would like to try out the new iOS and it crashes unless I uninstall the app.

**Comment by Roundlay** [comment]
+1… Awaiting TestFlight link. (:

**Comment by kspearrin** [comment]
Here is a public link to testflight with a build targeting iOS 12 minimum. Please report back if this resolves the problem.

https://testflight.apple.com/join/595lFH7U

**Comment by Nexus237** [comment]
> Here is a public link to testflight with a build targeting iOS 12 minimum. Please report back if this resolves the problem.
> 
> 
> 
> https://testflight.apple.com/join/595lFH7U

Thanks @kspearrin

**Comment by trparky** [comment]
Nope, still crashes on my iPhone 11 Pro.

**Comment by tomsinclair94** [comment]
Can confirm also still crashing on iPhone 12 Pro - TestFlight had 3 crash “popups” as if it tried to open multiple times

**Comment by takov751** [comment]
Same on iPhone 11 crash popup 3x

**Comment by dfuentes03** [comment]
iPhone 12 Pro Max on iOS 15 B6

The first launch of the application was successful. I then quit the application by swiping up on it in the app switcher and instantly got a "Bitwarden Crashed" popup. Any subsequent attempt to launch Bitwarden just results in the app crash popup being displayed.

**Comment by kspearrin** [comment]
Thanks all. We'll see if there is something else we can try.

**Comment by dfuentes03** [comment]
> Thanks all. We'll see if there is something else we can try.

Anything we can collect and get to you to facilitate determining root cause? Or is sending the TestFlight Application Crash report when we get the popup sufficient?

**Comment by Vladx71** [comment]
Still crashing with the latest public beta after the first successful start

L:

**Comment by karassm** [comment]
Same crash on iphone 12 ios 15b6

**Comment by trparky** [comment]
@kspearrin Someone mentioned something about LLVM over at the Xamarin project. Does that make any sense?

**Comment by trparky** [comment]
@kspearrin https://mallibone.com/post/custom-targets-llvm

**Comment by spencer-gre** [comment]
Thought I'd share that I no longer crash on B6 (meaning I can open the app) but the auto fill/keyboard extension is unusable.

**Comment by craigmpeters** [comment]
iPhone 10S Max
15.0 (19A5325f) 
initially worked including safari but first launch after using safari it now crashes each attempt to launch.

**Comment by kspearrin** [comment]
I just added build 2.12.0 (846) to the public testflight track. Please try this and report back. It has LLVM enabled as a suggested workaround.

**Comment by trparky** [comment]
Well, we have a new issue. Great. It crashes on exit, however... it launches after the app crashes whereas before, it didn't. One step forward, I guess.

**Comment by rbrussell82** [comment]
> 
> 
> I just added build 2.12.0 (846) to the public testflight track. Please try this and report back. It has LLVM enabled as a suggested workaround.

This seems to fix the app crashes for me on an iPhone 11 Pro Max, BUT AutoFill doesn't seem to be working.  It never opens the passwords for me or if it detects a saved login and I click on it, it doesn't do anything.

**Comment by nicopasla** [comment]
Build 846 work perfectly fine for me, auto fill is also working perfectly

**Comment by rbrussell82** [comment]
> 
> 
> > I just added build 2.12.0 (846) to the public testflight track. Please try this and report back. It has LLVM enabled as a suggested workaround.
> 
> This seems to fix the app crashes for me on an iPhone 11 Pro Max, BUT AutoFill doesn't seem to be working. It never opens the passwords for me or if it detects a saved login and I click on it, it doesn't do anything.

AutoFill does seem to be working for me after restarting the phone.  No crashes here after a phone reboot.

**Comment by trparky** [comment]
I rebooted my phone; the app still crashes on exit however the app works. Something about small victories.

**Comment by craigmpeters** [comment]
Updated build: auto fill worked once but then didn't, it currently does not crash when using the app after this which is different from my test done previously.

**Comment by MrAshieB** [comment]
Build 2.12.0 (846) no longer crashing on launch, but does crash on closing, prompting a pop-up to report to devs.

**Comment by takov751** [comment]
> Build 2.12.0 (846) no longer crashing on launch, but does crash on closing, prompting a pop-up to report to devs.

Exactly the same 👌 so it seems the solution is in reach

**Comment by oaklandm** [comment]
Yep seems to have fixed the crash on launch, thanks for finding the fix. I haven't experienced a crash on exit, or any crashes at all

**Comment by manikantareddyd** [comment]
Is there scope for some conspiracy theory here?

**Comment by kspearrin** [comment]
We just uploaded build 2.12.0 (848) that might fix also the autofill extensions. Please let us know.

**Comment by manikantareddyd** [comment]
Can confirm that with build 2.12.0 (848) both Autofill and App work without crashing on iPhone 12 pro max with 15.0 Beta 5 (19A5325f).

**Comment by ethanhassett** [comment]
I can confirm 848 is also working correctly on my iPhone 12 Pro on iOS 15 Beta 6 (19A5325f).

**Comment by josephazrak** [comment]
Can confirm that the application now functions without crashing as of TF build 848. Auto fill works too.

App crashes on task switcher exit, though.

**Comment by karassm** [comment]
With build 848 on iphone 12 ios15b6 : tested ok (app + autofill) on safari, chrome and firefox focus... no crash yet. Thanks a lot for the availability on TestFlight !

**Comment by nicknitewolf** [comment]
Steps to reproduce crash on exit:

1) Open bitwarden
2) Open recents
3) Close bitwarden from recents
4) Bitwarden crashes

It is crucial to not go back to the homescreen before closing Bitwarden as that scenario is working fine

**Comment by dfuentes03** [comment]
Thanks to the Bitwarden team for hearing their users and getting this fixed even if on iOS Beta software! Another reason to continue using and paying for your product.

**Comment by hv0905** [comment]
Build 848 works well on ios15 beta6（iphone8plus）
Thanks a lot for the fix!

**Comment by egadsthefuzz** [comment]
> Steps to reproduce crash on exit:
> 
> 
> 
> 1) Open bitwarden
> 
> 2) Open recents
> 
> 3) Close bitwarden from recents
> 
> 4) Bitwarden crashes
> 
> 
> 
> It is crucial to not go back to the homescreen before closing Bitwarden as that scenario is working fine

I can reproduce on 848 using this, autofill now works though

**Comment by nicknitewolf** [comment]
For people still looking for the testflight link:

https://testflight.apple.com/join/595lFH7U

**Comment by xiangsanliu** [comment]
> > Steps to reproduce crash on exit:
> > 
> > 1. Open bitwarden
> > 2. Open recents
> > 3. Close bitwarden from recents
> > 4. Bitwarden crashes
> > 
> > It is crucial to not go back to the homescreen before closing Bitwarden as that scenario is working fine
> 
> I can reproduce on 848 using this, autofill now works though

Same problem on 848 and iOS15 beta.

**Comment by c0ldplasma** [comment]
Can confirm the AppStore version works on iPhone 12 iOS 15 Beta 7 without crashing
